tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -823,7 +886,15 @@ public:
    {
       return Detail::RFieldValue(this, static_cast<T *>(where), std::forward<ArgsT>(args)...);
    }
-   ROOT::Experimental::Detail::RFieldValue GenerateValue(void *where) final { return GenerateValue(where, T()); }
+   ROOT::Experimental::Detail::RFieldValue GenerateValue(void *where) final
+   {
+      if constexpr (std::is_default_constructible_v<T>) {
```

Comment:
This is nifty :).  One unfortunate downside :( ... By definition if a user created a `TRootIOCtor` constructor (even if it also has a default constructor) most likely mean for that constructor to be used.  So we actually need:
```
if (has_TRootIOCtor) 
    ....
else if constexpr (std::is_default_constructible_v<T>) 
   ....
else 
   fail ....